src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,117 @@
+package duke.command;
+
+import duke.exception.IncompleteDescriptionException;
+import duke.exception.InvalidDateFormatException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/** A class that handles task-addition command. */
+public class AddCommand extends Command{
+
+    private enum TaskType {
+        TODO, EVENT, DEADLINE
+    }
+
+    private final String taskType;
+    private final String taskDescriptions;
+
+    /**
+     * A constructor for class AddCommand.
+     *
+     * @param taskType The type of the to-be-added task.
+     * @param taskDescriptions The description of the to-be-added task.
+     */
+    public AddCommand(String taskType, String taskDescriptions) {
+
+        this.taskType = taskType;
+        this.taskDescriptions = taskDescriptions;
+    }
+
+    private String[] validateCommand()
+            throws IncompleteDescriptionException,
+            InvalidDateFormatException {
+
+        String[] descriptionComponents = this.taskDescriptions.split(" /by | /at ", 2);
+        String descriptionMessage = "The description of a %s is incomplete.";
+        String dateMessage = "Please specify the date in yyyy-mm-dd format!";
+
+        if (this.taskType.equals("TODO")) {
+            return descriptionComponents;
+        }
+
+        if (descriptionComponents.length < 2 || descriptionComponents[1].trim().isEmpty()) {
+            throw new IncompleteDescriptionException(String.format(descriptionMessage, this.taskType.toLowerCase()));
+        }
+
+        if (!descriptionComponents[1].matches("\\d{4}-\\d{2}-\\d{2}")) {
+            throw new InvalidDateFormatException(dateMessage);
+        }
+
+        return descriptionComponents;
+    }
+
+    private Task createTask(String description, LocalDate date) {
+
+        switch (TaskType.valueOf(this.taskType)) {
```

Comment:
I believe you should avoid unnecessary use of `this`. Instead, you could just write `TaskType.valueOf(taskType)`. This applies to your other codes as well :)